moving server options to references (#1549)
temporalio/documentation: docs/references/server-options.md

---
id: server-options
title: Temporal Server options
description: undefined
sidebar_label: Start options
---

<!-- This file is generated. Do not edit it directly. -->

You can run the Temporal server as a Go application by including the server package `go.temporal.io/server/temporal` and using it to create and start a new Temporal Server.

> Note, there are many ways to [run a Temporal Cluster](/kb/install-temporal-cluster-for-development) and this is approach is recommended for a limited number of situations.

Reviewer: ```suggestion
Note, there are many ways to [run a Temporal Cluster](/kb/install-temporal-cluster-for-development) and this approach is recommended for a limited number of situations.
```

```go
s := temporal.NewServer()
err := s.Start()
if err != nil{
	log.Fatal(err)
}
```

`NewServer()` accepts functions as parameters.
Each function returns a `ServerOption` that is applied to the instance.
Source code for parameter reference is here: https://github.com/temporalio/temporal/blob/master/temporal/server_option.go

### WithConfig

Overrides the default configuration with a custom configuration that is defined in the config package `go.temporal.io/server/common/service/config`.

```go
s := temporal.NewServer(
	temporal.WithConfig(cfg),
)
```

### WithConfigLoader

Load a custom configuration from a file.

```go
s := temporal.NewServer(
	temporal.WithConfigLoader(configDir, env, zone),
)
```

### ForServices

Sets the list of all valid temporal services.
The default can be used from the `go.temporal.io/server/temporal` package.

```go
s := temporal.NewServer(
	temporal.ForServices(temporal.Services),
)
```

### InterruptOn

This option provides a channel that interrupts the server on the signal from that channel.

> - If `temporal.InterruptOn()` is not passed, `server.Start()` is not blocking and `server.Stop()` needs to be called somewhere.

Reviewer: ```suggestion
- If `temporal.InterruptOn()` is not passed, `server.Start()` is never blocked and you need to call `server.Stop()` somewhere.
```

- If `temporal.InterruptOn(nil)` is passed, `server.Start()` blocks forever until the process is killed.

> - If `temporal.InterruptOn(temporal.InterruptCh())` is passed, `server.Start()` blocks until Ctrl+C is used and then the server is gracefully shut down.

Reviewer: ```suggestion
- If `temporal.InterruptOn(temporal.InterruptCh())` is passed, `server.Start()` blocks until you use Ctrl+C, which then gracefully shuts the server down.
```
